include/envoy/http/access_log.h

```diff
@@ -114,6 +114,16 @@ class RequestInfo {
    * Set whether the request is a health check request or not.
    */
   virtual void healthCheck(bool is_hc) PURE;
+
+  /**
+   * Set the downstream address.
+   */
+  virtual void setDownstreamAddress(const std::string& address) PURE;
```

Comment:
@khgandhi you can remove this from the interface and directly set downstream_address_ (as it's a public member of a struct). Similar to bytesReceived: https://github.com/lyft/envoy/blob/master/source/common/http/access_log/request_info_impl.h#L51
https://github.com/lyft/envoy/blob/master/source/common/http/access_log/request_info_impl.h#L19